ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java

```diff
@@ -93,7 +99,9 @@ static Schema getSchemaFromType(final Type type, final String name, final String
       schema = GenericsUtil.generic(((TypeVariable) type).getName());
     } else {
       schema = typeToSchema.getOrDefault(type, () -> handleParametrizedType(type)).get();
-      schema.name(name);
+      if (schema.name() == null) {
```

Comment:
Why is this change necessary? (Why would the schema already have a name at this point?)

Reply:
this is something that I feel like we should change in the long run, but decimals are defined by their schema name (they all have `org.apache.kafka.connect.data.Decimal` as their schema name). we should not be using the schema name to define the parameter name, but that's out of scope

Reply:
Ah, my bad. I even followed the Decimal schema builder code through but somehow missed it was setting the name. Thanks for the explanations!